src/main/java/Control/Scenario/UC_FlatSummary.java

```diff
@@ -0,0 +1,175 @@
+                project.setCreatedat(tp.getA_CreatedAt());
+                project.setDeletedat(tp.getA_DeletedAt());
+
+                temp.add(project);
+            }
+
+            db.afterOkSqlExecution();
+
+            // dont forget to set data that was inserted into json
+            projects.setProjects(temp);
+
+        } catch (SQLException sqle) {
+            db.afterExceptionInSqlExecution(sqle);
+        }
+
+        return projects;
+    }
+
+    /***
+     * Returns null if he has no right to see measurements for that flat or it does not exist
+     * @param userId from sesssion
+     * @param flatId from request
+     * @return
+     */
+    public ArrayList<T_Measurement> getAllMeasurementsForFlat(int userId, int flatId) {
+        if (doesUserHaveRightToSeeProjectBelongingToFlat(userId, flatId) == false) {
+            return null;
+        }
+
+        //
+        // has right to see flat, return empty at the very least
+        //
+        ArrayList<T_Measurement> measurementsArr = new ArrayList<T_Measurement>();
+
+        // Retrieve all sensor that belong to the controllers that belong to the flat
+        ArrayList<T_ControllerUnit> controllerArr = getAllControllersForFlat(flatId);
+        ArrayList<T_Sensor> sensorsArr = new ArrayList<T_Sensor>();
+
+        for (T_ControllerUnit tcu: controllerArr
+             ) {
+            sensorsArr.addAll(getAllSensorsForController(tcu.getA_pk()));
+        }
+
+        // add all measurements to resulting array
+        for(T_Sensor ts: sensorsArr) {
+            measurementsArr.addAll(getAllMeasurementsForSensor(ts.getA_pk()));
+        }
+
+        return measurementsArr;
+    }
+
+    /***
+     * Temporary //TODO remove
+     * @param userId
+     * @param flatId
+     * @return
+     */
+    public boolean temporaryCheckToSeeIfHasRightToSee(int userId, int flatId) {
+        return doesUserHaveRightToSeeProjectBelongingToFlat(userId, flatId);
+    }
+
+    /// PRIVATES
+
+    /***
+     * Important function, do not change
+     * Verifies rights for user's right to view flat (that belongs to certain project)
+      * @param userId from session
+     * @param flatId flat that is attempting to see
+     * @return
+     */
+    private boolean doesUserHaveRightToSeeProjectBelongingToFlat(int userId, int flatId) {
+        boolean toReturn = false;
+
+        try {
+            db.beforeSqlExecution();
+
+            if (null != GeneralAccessibility.doesUserHaveRightToAccessFlat(db.getConn(), db.getPs(), db.getRs(), userId, flatId)) {
+                toReturn = true;
+            }
+
+            db.afterOkSqlExecution();
+        } catch (SQLException sqle) {
+            db.afterExceptionInSqlExecution(sqle);
+        }
+
+        return toReturn;
+    }
+
+    private ArrayList<T_ControllerUnit> getAllControllersForFlat(int flatId) {
+        ArrayList<T_ControllerUnit> arr = new ArrayList<T_ControllerUnit>();
+
+        try {
+            db.beforeSqlExecution();
```

Comment:
Transakcia ma ist  podla mna ked tak v celej public metode kontrolera (o 1 vyssie), nie ze volat to v kazdom podvolani na databazu. Navyse tu napr. nechapem ani ze preco sa vola ked je medzi nimi len 1 riadok.

Reply:
Tam kde sa robia vkladacky to samozrejme ma byt transakcia na celok a nie na jednotlivcov. Toto je vsak aj na logovanie a zistovanie problemov v jednotlivych castiach.